### DISCUSSION TITLE
Read From Zipped Folder

### DISCUSSION THREAD

**Discussion by nickboxx** [nextgenhealthcare/connect#5694] [Q&A]
I have a Channel Reader source with some javascipt in it that is going to the appliance and is reading from a directory until it finds a zipped folder in it.

What I am trying to do is go in to that zipped folder and read the contents from it however I am having issues even getting the list of files from the zipped folder. My testing channelMap 'aaaa' just outputs null. Is this even possible to do?

        if(fileName.includes('.zip')){
	        var inboxPath2 = files[fileIdx]
	        var inboxDirectory2 = new java.io.File(inboxPath2)
	        channelMap.put('inboxDirectory2', inboxDirectory2)
	        
	        channelMap.put('aaaa', inboxDirectory2.list())
        
        }

**Comment by jonbartels** [comment]
`channelMap.put('aaaa', inboxDirectory2.list()` is the same operation as `ls` or `dir` on the command line. You need to list the contents of the ZIP file, it is not a directory.

You need to use something like Javas native `ZipFile` classes or https://commons.apache.org/proper/commons-compress/zip.html

**Comment by rjkroll** [comment]
Here is a quick example of reading in the contents of a ZIP that you can use in a Javascript transformer.    Tested on MC 4.2/OpenJDK 19:

```
// Set the path to the folder containing the zip file
var folderPath = "/path/to/folder";

// Set the name of the zip file
var zipFileName = "example.zip";

// Create a File object for the zip file
var zipFile = new Packages.java.io.File(folderPath + "/" + zipFileName);

// Create a ZipInputStream to read the contents of the zip file
var zipIn = new Packages.java.util.zip.ZipInputStream(new Packages.java.io.FileInputStream(zipFile));

// Loop through the zip file entries and read the contents of each file
var entry = zipIn.getNextEntry();
while (entry != null) {
    var fileName = entry.getName();
    var fileContents = new Packages.org.apache.commons.io.IOUtils.toString(zipIn);
    // Do something with the file contents, such as logging or parsing
    logger.info("File name: " + fileName);
    logger.info("File contents: " + fileContents);
    entry = zipIn.getNextEntry();
}

// Close the ZipInputStream
zipIn.close();
```
